core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -4504,19 +4504,25 @@ private Metrics() {} // prevent instantiation
    * @return newly-created {@link FileSystemMasterAuditContext} instance
    */
   private FileSystemMasterAuditContext createAuditContext(String command, AlluxioURI srcPath,
-      @Nullable AlluxioURI dstPath, @Nullable Inode srcInode)
-      throws AccessControlException {
+      @Nullable AlluxioURI dstPath, @Nullable Inode srcInode) {
     FileSystemMasterAuditContext auditContext =
         new FileSystemMasterAuditContext(mAsyncAuditLogWriter);
     if (mAsyncAuditLogWriter != null) {
-      String user = AuthenticatedClientUser.getClientUser(ServerConfiguration.global());
-      String ugi;
+      String user = null;
+      String ugi = "";
       try {
-        String primaryGroup = CommonUtils.getPrimaryGroupName(user, ServerConfiguration.global());
-        ugi = user + "," + primaryGroup;
-      } catch (IOException e) {
-        LOG.warn("Failed to get primary group for user {}.", user);
-        ugi = user;
+        user = AuthenticatedClientUser.getClientUser(ServerConfiguration.global());
+      } catch (AccessControlException e) {
+        ugi = "N/A";
+      }
+      if (user != null) {
+        try {
+          String primaryGroup = CommonUtils.getPrimaryGroupName(user, ServerConfiguration.global());
+          ugi = user + "," + primaryGroup;
+        } catch (IOException e) {
+          LOG.warn("Failed to get primary group for user {}.", user);
```

Comment:
will this show up once or on each rpc?

Reply:
It can show up on each RPC, which might be too chatty. Do you want to make it a debug log?

Reply:
When are we supposed to see the exception to thrown? It looks like we are throwing it from `alluxio.security.group.CachedGroupMapping#getGroups` which we may return an empty list instead.
If we expect one exception per RPC in the target use case, this is very very expensive and Master process will become very slow.

If we don't throw exception, maybe use just use "N/A" in ugi.group to indicate the failure to get group?

Reply:
makes sense. done.